pkg/internal/vault/vault.go

```diff
@@ -273,6 +273,13 @@ func (v *Vault) requestTokenWithAppRoleRef(client Client, appRole *v1.VaultAppRo
 		return "", fmt.Errorf("error encoding Vault parameters: %s", err.Error())
 	}
 
+	vaultIssuer := v.issuer.GetSpec().Vault
```

Comment:
The namespace is supported as part of the API route - added after the "v1" fragment in the URI. It is a strongly typed solution than header based option. A simpler fix is to make this namespace check and append the vaultIssuer.namespace to the ```.Join``` call in ```Line 267```.

Reply:
@krishnan-recrooz My business need requires that the vault namespace be sent via an HTTP header.  This is how we have configured our vault servers to handle namespaced API calls.  If the vault namespace is sent via the URL, my requests to my vault server will default to the root namespace and this feature will not work as I expect it.  This code should solve that requirement with the X-VAULT-NAMESPACE header.

Reply:
This pattern is also followed elsewhere by Vault clients so this is probably the right fix.